packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -143,6 +143,11 @@ export class ViewerHotSpotResult {
      * 3D world coordinates
      */
     public readonly worldPosition: [x: number, y: number, z: number] = [NaN, NaN, NaN];
+
+    /**
+     * visibility range is [-1..1]. A value of 0 means camera eye is on the plane.
+     */
+    public visibility: number = NaN;

```

Comment:
I'm wondering if we should include the raw worldNormal as a property, and then have visibility be a computed property, since I ended up switching `ViewerHotSpotResult` back to a class for other reasons anyway. We could change this in the future, but if we did it this way then `visibility` should be readonly (e.g. only have a getter) so technically it would be a small breaking change if we do it later.

Reply:
Meh, I think we can just change it later if needed. It's already a mostly mutable object anyway that exists for GC efficiency.